app/services/waste_exemptions_engine/edit_completion_service.rb

```diff
@@ -19,8 +19,6 @@ def find_original_registration
     end
 
     def copy_data_from_edit_registration
-      related_objects_changed?
```

Comment:
I am totally unaware of why this code was here, and hence I am going to wait for Iris feedback before merging this code

Reply:
Ok so it took me a while, and failing tests helped, but I figured. 
The issue is that when copying related objects in the `@registration` we are effectively saving the registration. Hence when then comparing the object at the end no changes arise. 
In order to fix this, I have changed the logic a bit, by checking first of all if the object has changed at all, and if it did then I do the copy over. In the end, I always create a new version.